src/command/subgraph/publish.rs

```diff
@@ -42,7 +52,49 @@ impl Publish {
         client_config: StudioClientConfig,
         git_context: GitContext,
     ) -> RoverResult<RoverOutput> {
+        // * if --allow-invalid-routing-url is not provided
+        // * AND a --routing-url is provided
+        // * AND the URL is unparsable
+        // we need to warn and prompt the user, else we can assume a publish
+        if !self.allow_invalid_routing_url {
+            if let Some(routing_url) = &self.routing_url {
+                if let Err(parse_error) = Url::parse(routing_url) {
+                    tracing::debug!("Parse error: {}", parse_error.to_string());
+
+                    if let Some(result) =
+                        Self::handle_invalid_routing_url(Self::prompt_for_publish(), None)
+                    {
+                        return Ok(result);
+                    }
+                }
+            }
+        }
+
+        // below is borrowed heavily from the `Fetch` command
         let client = client_config.get_authenticated_client(&self.profile)?;
+
+        if self.routing_url.is_none() {
```

Comment:
this `if self.routing_url.is_none()` should probably just be the `else` block on your first `if let Some(routing_url) = &self.routing_url` block